### PR TITLE
Reminder of where the celltoolbar menu is

### PR DESCRIPTION
**Pull request by Carreau**
The UI has changed since 4.0 so 4.1 need to tell that to users.

Ping @ellisonbg , @jdfreder 

ct

### CODE REVIEW

**Comment by Carreau** [comment]
@dsblank @jhamrick is it worth abstracting into a util function that would be useful for teaching where the menus are ?

**Comment by jdfreder** [comment]
This is a cool idea.  I don't think I've ever seen another program do something like this, it's a deprecation warning for our UI.   However, we should definitely remove this button for 4.2 or 5.0- I don't think it makes sense to have it linger around forever.

File: notebook/static/notebook/js/maintoolbar.js [hunk 1]
```diff
@@ -55,10 +55,34 @@ define([
...
+
+        _b.on('click', function(){
+            setTimeout(function(){$('#view_menu').parent().addClass('pulse')},0) 
+            setTimeout(function(){$('#view_menu').parent().addClass('open')},1000) 
+            setTimeout(function(){$('#menu-cell-toolbar').children('a').addClass('pulse')},2000)
+            setTimeout(function(){$('#menu-cell-toolbar').children('ul').css('display','block')},3000)
+            setTimeout(function(){$('#menu-cell-toolbar').children('ul').css('display','')},5400)
+            setTimeout(function(){$('#menu-cell-toolbar').children('a').removeClass('pulse')},5600)
+            setTimeout(function(){$('#view_menu').parent().removeClass('open')},5800)
+            setTimeout(function(){$('#view_menu').parent().removeClass('pulse')},6000)
```
Comment (jdfreder) [inline notebook/static/notebook/js/maintoolbar.js]: The time intervals aren't consistent, was that intentional?

Comment (Carreau) [inline notebook/static/notebook/js/maintoolbar.js]: Yes that "remove" part is faster. I played a bit for the menu appearance to look "natural", and for that the disappear part need to be faster than the appear part (like when you remove your mouse).

If you look closely, each step of opening the menu is 1s, it stays open for 2.4 and disappear by step of 200ms.

**Comment by Carreau** [comment]
I think we can even do something nicer with @sccolbert's phosphor menus, I just don't want to implement it now if we will have better menus in a few month. 

In particular I think we can use a 2cd color to mimic the fact the the mouse hovers elements it will not click onto, like what you see when I actually use my mouse. 

Note that in the GIF the menu disappearing might look irregular, but that's just because the GIF is at 8 fps.

**Comment by ellisonbg** [comment]
I like this idea for deprecating UI. We won't always be able to do
something like this, but when we can, it is great.

I haven't looked at the code, but love the idea and demo above.

On Sat, Oct 24, 2015 at 4:42 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> I think we can even do something nicer with @sccolbert
> <https://github.com/sccolbert>'s phosphor menus, I just don't want to
> implement it now if we will have better menus in a few month.
>
> In particular I think we can use a 2cd color to mimic the fact the the
> mouse hovers elements it will not click onto, like what you see when I
> actually use my mouse.
>
> Note that in the GIF the menu disappearing might look irregular, but
> that's just because the GIF is at 8 fps.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/651#issuecomment-150849427>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by minrk** [comment]
Nice. It seems a bit like the button should disappear after showing where its new home is. Or should it stick around?

**Comment by Carreau** [comment]
it seem like if we make it disappear, we would need to persist the hidden state. 
Which I don't want to start doing.

**Comment by minrk** [comment]
Fair enough - this is probably a good compromise for a short-lived deprecation.

**Comment by minrk** [comment]
Works as advertised.

**Pull request merged**